node-admin/src/test/java/com/yahoo/vespa/hosted/node/admin/nodeagent/NodeAgentContextManagerTest.java

```diff
@@ -141,51 +141,43 @@ public void setFrozen_is_successful_if_converged_in_time() throws InterruptedExc
         assertEquals(Optional.of(true), asyncScheduler.response);
     }
 
+    private void waitUntilWaitingForNextContext() {
```

Comment:
This should be a method on NodeAgentContext using Object::wait and notify with no sleep necessary.